### PR TITLE
Integrate Copilot SDK

### PR DESCRIPTION
**Pull request by sajjanstha**
#### The main changes are:

- [x] Install Copilot SDK
- [x] Fetch token from param and send to the services

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -2,3 +2,5 @@ COPILOT_API_KEY=""
 POSTGRES_PRISMA_URL=""
 POSTGRES_URL_NON_POOLING=""
 BLOB_READ_WRITE_TOKEN=""
+COPILOT_ENV="local"
+API_URL="http://localhost:3000/"
```
Comment (sazanrjb) [inline .env.example:6]: do we need this env variable? Where is it used?

Comment (sajjanstha) [reply .env.example:6]: It is being used in the frontend to call our settings, media, etc endpoints

Comment (sazanrjb) [inline .env.example:6]: @aatbip I think we replaced this with VERCEL_URL, so we can remove this

File: src/app/api/client/route.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+  try {
+    const client = await copilotClient.getClient(z.string().parse(clientId))
 
-  const res = await fetch(`${copilotAPIUrl}/v1/clients/${clientId}`, options)
-  const client = await res.json()
-  return NextResponse.json({ data: client })
+    return NextResponse.json({ data: client })
+  } catch (error) {
+    console.log(error)
+    return NextResponse.json({ data: {} })
```
Comment (sazanrjb) [inline src/app/api/client/route.ts:23]: we should send error response..

File: src/app/api/clients/route.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  const copilotClient = new CopilotAPI(z.string().parse(token))
+  try {
+    const clients = await copilotClient.getClients()
+
+    return NextResponse.json(clients)
+  } catch (error) {
+    console.log(error)
+    return NextResponse.json({ data: {} })
```
Comment (sazanrjb) [inline src/app/api/clients/route.ts:20]: Same here, we should send the error

File: src/app/client-preview/page.tsx [hunk 4]
```diff
@@ -5,32 +5,41 @@ import { apiUrl } from '@/config'
...
+  searchParams,
+}: {
+  searchParams: { token: string }
+}) {
+  const { token } = searchParams
+  const settings = await getSettings(token)
 
   //the clientId will be dynamic later
-  const _client = await getClient('a2b66ac1-eedb-4b1c-8b6e-941b71583f3f')
+  const _client = await getClient('80ca5645-73e8-4a22-9bf5-cc5327355972', token)
```
Comment (sazanrjb) [inline src/app/client-preview/page.tsx:40]: we should call .me endpoint to get current client user and use the id from that

File: src/app/components/EditorInterface.tsx [hunk 5]
```diff
@@ -227,25 +233,24 @@ const EditorInterface = () => {
...
   useEffect(() => {
     if (!appState?.appState.settings) return
-    setTimeout(() => {
-      editor
-        ?.chain()
-        .focus()
-        .setContent((appState?.appState.settings as ISettings).content)
-        .run()
-    })
+    // setTimeout(() => {
```
Comment (sazanrjb) [inline src/app/components/EditorInterface.tsx:247]: can be removed?

File: src/app/page.tsx [hunk 6]
```diff
@@ -19,34 +15,39 @@ async function listClients() {
...
     throw new Error('Something went wrong while fetching client list!')
   }
 
-  const { data } = await res.json()
+  const { autofillFields } = await res.json()
+  return autofillFields
+}
+
+async function getSettings(token: string) {
+  const { data } = await fetch(`${apiUrl}/api/settings?token=${token}`).then(
```
Comment (sazanrjb) [inline src/app/page.tsx:30]: lets use either await  or .then and not mix them

File: src/components/tiptap/autofieldSelector/autofillMenuSuggestion.ts [hunk 7]
```diff
@@ -2,20 +2,26 @@ import { ReactRenderer } from '@tiptap/react'
...
+async function getCustomFields(token: string) {
+  const res = await fetch(`/api/autofill?token=${token}`, {
     next: { revalidate: 0 },
   })
   const { autofillFields } = await res.json()
 
   return autofillFields
 }
 
+const token = new URLSearchParams(document.location.search).get('token')
```
Comment (sazanrjb) [inline src/components/tiptap/autofieldSelector/autofillMenuSuggestion.ts:16]: may be better to validate here

File: src/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx [hunk 8]
```diff
@@ -72,7 +72,13 @@ export const floatingMenuSuggestion = {
           tiptapEditorUtils.deleteRange(range)
           const imagePickerUtils = new ImagePickerUtils()
           const file = await imagePickerUtils.selectImageFromLocalDrive()
-          const data = await handleBannerImageUpload(file as File)
+          const token = new URLSearchParams(document.location.search).get(
```
Comment (sazanrjb) [inline src/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx:75]: val

File: src/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx [hunk 9]
```diff
@@ -72,7 +72,13 @@ export const floatingMenuSuggestion = {
...
+            'token',
+          )
+          const data = await handleBannerImageUpload(
+            file as File,
+            token as string,
```
Comment (sazanrjb) [inline src/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx:80]: validate instead of cast

File: src/app/client-preview/page.tsx (hunk 4 above)
Comment (aatbip) [reply src/app/client-preview/page.tsx:40]: @sajjanstha I think we should add another API for this.

File: .env.example (hunk 1 above)
Comment (aatbip) [reply .env.example]: I have removed this in the latest commit

File: src/app/client-preview/page.tsx (hunk 4 above)
Comment (aatbip) [reply src/app/client-preview/page.tsx]: integrated api/me api to the preview route

File: src/app/components/EditorInterface.tsx (hunk 5 above)
Comment (aatbip) [reply src/app/components/EditorInterface.tsx]: removed in the latest commit

File: src/app/page.tsx (hunk 6 above)
Comment (aatbip) [reply src/app/page.tsx]: replaced with async/await in latest commit

**Review by sazanrjb — APPROVED** [review]

**Pull request merged**